### PR TITLE
fix: correct mapping from exception to status, cleanup, tests

### PR DESCRIPTION
**Pull request by zoltanpalfi**
Part of AAE-1503

### CODE REVIEW

File: activiti-cloud-services-runtime-bundle/activiti-cloud-services-rest-impl/src/main/java/org/activiti/cloud/services/rest/controllers/RuntimeBundleExceptionHandler.java [hunk 1]
```diff
@@ -1,55 +1,60 @@
...
-    }
-
     @ExceptionHandler(ActivitiInterchangeInfoNotFoundException.class)
-    @ResponseStatus(HttpStatus.NO_CONTENT)
+    @ResponseStatus(NO_CONTENT)
     public Resource<ActivitiErrorMessage> handleAppException(ActivitiInterchangeInfoNotFoundException ex, HttpServletResponse response) {
-        response.setContentType(MediaType.APPLICATION_JSON_VALUE);
-        return new Resource<>(new ActivitiErrorMessageImpl(HttpStatus.NOT_FOUND.value(), ex.getMessage()));
+        response.setContentType(APPLICATION_JSON_VALUE);
+        return new Resource<>(new ActivitiErrorMessageImpl(NO_CONTENT.value(), ex.getMessage()));
```
Comment (ffazzini) [inline activiti-cloud-services-runtime-bundle/activiti-cloud-services-rest-impl/src/main/java/org/activiti/cloud/services/rest/controllers/RuntimeBundleExceptionHandler.java]: if it is an error, it a not found right?

Comment (ffazzini) [inline activiti-cloud-services-runtime-bundle/activiti-cloud-services-rest-impl/src/main/java/org/activiti/cloud/services/rest/controllers/RuntimeBundleExceptionHandler.java]: it was not found before.. but is it no content an error code?

Comment (zoltanpalfi) [reply activiti-cloud-services-runtime-bundle/activiti-cloud-services-rest-impl/src/main/java/org/activiti/cloud/services/rest/controllers/RuntimeBundleExceptionHandler.java]: No, the initial code was mapping the exception to no content (204), which as you highlighted, is not an error code. I didn't changed that, merely just in the response body I was returning the same status.

I do think this should map to some error status indeed, but that change needs a bit investigation

Comment (ffazzini) [reply activiti-cloud-services-runtime-bundle/activiti-cloud-services-rest-impl/src/main/java/org/activiti/cloud/services/rest/controllers/RuntimeBundleExceptionHandler.java]: ok, i would suggest to keep the investigation about this specific change on a different pr. And to keep this pr  only for the 422 introduction on that api only.

Comment (zoltanpalfi) [reply activiti-cloud-services-runtime-bundle/activiti-cloud-services-rest-impl/src/main/java/org/activiti/cloud/services/rest/controllers/RuntimeBundleExceptionHandler.java]: Okay, reverted, created issue for that to be checked: https://github.com/Activiti/activiti-cloud-runtime-bundle-service/issues/504

**Review by mteodori — APPROVED** [review]
lgtm

**Review by miguelruizdev — APPROVED** [review]
LGTM

**Comment by zoltanpalfi** [comment]
@Mergifyio refresh

**Comment by erdemedeiros** [comment]
@Mergifyio refresh

**Pull request merged**